sway-core/src/parse_tree/use_statement.rs

```diff
@@ -38,62 +37,136 @@ impl UseStatement {
         } else {
             stmt.clone().next().expect("Guaranteed by grammar")
         };
-        let mut import_path_buf = vec![];
-        let mut import_path_vec = import_path.into_inner().collect::<Vec<_>>();
-        let last_item = import_path_vec.pop().unwrap();
-        let import_type = match last_item.as_rule() {
-            Rule::star => ImportType::Star,
-            Rule::ident => ImportType::Item(check!(
-                Ident::parse_from_pair(last_item, config),
+
+        let use_statements_buf = check!(
+            handle_import_path(import_path, config, is_absolute),
+            return err(warnings, errors),
+            warnings,
+            errors
+        );
+
+        ok(use_statements_buf, Vec::new(), Vec::new())
+    }
+}
+
+fn handle_import_path(
+    import_path: Pair<Rule>,
+    config: Option<&BuildConfig>,
+    is_absolute: bool,
+) -> CompileResult<Vec<UseStatement>> {
+    let mut errors = vec![];
+    let mut warnings = vec![];
+
+    // Populate the call path
+    let mut import_path_buf = vec![];
+    let mut import_path_vec = import_path.clone().into_inner().collect::<Vec<_>>();
+    let mut last_item = import_path_vec.pop().unwrap();
+
+    let mut top_level_alias = None;
+    if last_item.as_rule() == Rule::alias {
+        let item = last_item.into_inner().nth(1).unwrap();
+        let alias_parsed = check!(
+            Ident::parse_from_pair(item, config),
+            return err(warnings, errors),
+            warnings,
+            errors
+        );
+        top_level_alias = Some(alias_parsed);
+        last_item = import_path_vec.pop().unwrap();
+    }
+
+    for item in import_path_vec.into_iter() {
+        if item.as_rule() == Rule::ident {
+            import_path_buf.push(check!(
```

Comment:
This is now guaranteed by the language actually. Every `item` should be an `ident`. There is probably no need for the `if`.